sycl/doc/extensions/experimental/sycl_ext_intel_fp_control.asciidoc

```diff
@@ -0,0 +1,171 @@
+= sycl_ext_intel_fp_control
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 7 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties]
+* link:../experimental/sycl_ext_oneapi_kernel_properties.asciidoc[
+  sycl_ext_oneapi_kernel_properties]
+
+== Status
+
+This is an experimental extension specification, intended to provide early access
+to features and gather community feedback. Interfaces defined in this specification
+are implemented in {dpcpp}, but they are not finalized and may change incompatibly in
+future versions of {dpcpp} without prior notice. **Shipping software products should not
+rely on APIs defined in this specification.**
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices and
+only when using the Level Zero backend or OpenCL backend. Also currently Intel GPU
+driver supports this extension only for Vector Backend, so feature is supported only
+for ESIMD.
```

Comment:
I have couple of comments here:

* We should be more clear about when this extension works.  Is it only for kernels marked `SYCL_ESIMD_KERNEL`?  If so, let's say that.

* How hard is it to issue an error if the property is used with a kernel that is not marked `SYCL_ESIMD_KERNEL`?

Reply:
The only way I  was able to find is to emit that error in the frontend, added the test + updated documentation per your suggestions.